### ISSUE TITLE
Editing a capture to add bindings required re-entering password

### ISSUE THREAD

**Issue by dyaffe** [estuary/ui#496]
**Bug**
1. Create a PostgreSQL Capture
2. Edit it to add bindings
3. Get prompted to re-enter password

**Expected**
You shouldn't have to re-enter your password.

**Comment by kiahna-tucker** [comment]
This bug is more _nuanced_ than initially described (ref: [reported session](https://app.logrocket.com/tue20y/estuarytest/s/5-78aa2db7-a01d-4ec1-8c86-1d3439653799/13?cursor=1677104511497&end=1677103847221.1897&fromTab=sessions&start=1677103779226.5244&t=1677103847221.1897)). In production, I am able to rediscover a capture without issue and make edits to the captures bindings without being prompted to re-enter endpoint config information[^1].

(Skip to tl;dr if you want the gist.)

What I find interesting about the reported session:

* An error icon appears in the details form section header, but the page-level error only identifies an error in the endpoint config section of the form. In my experience, an error icon has only appeared in the details section form header when there is a permissions-related issue. Not saying that is the case here since it looks like the capture was able to be published but sharing for the record.

* An error icon does not appear in the endpoint config form section header despite the alleged presence of an error.

Given this bug cannot be simply recreated by altering bindings (i.e., resource config changes, single binding deletion, total binding deletion, single binding addition,  multi-binding addition, etc.) or rediscovering the capture in production, I am led to believe this is more of an edge case. Consequently, there should be a lower risk of a user encountering this defect.

### TL;DR
This bug is not as simple as originally believed at the time it was encountered. A deeper investigation is needed to find the root and the reproduction steps need to be updated so they reflect what is actually required to recreate this behavior consistently.

[^1]: The captures I have tested at this point in time have had a relatively low number of bindings discoverable (less than 10).

**Comment by kiahna-tucker** [comment]
On Feb. 23rd at 2:09PM EST, both myself and Dave were unable to reproduce this bug. If it is ever observed in the future, this issue should be reopened.
